### PR TITLE
Start of "reset" command with basic tests.

### PR DESCRIPTION
**Pull request by programaths**
Inspired by existing code and PR45

I added basic tests and they succeed. (distinction between "--soft" and "--hard")

### CODE REVIEW

File: src/GitElephant/Command/ResetCommand.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    /**
+     * @param TreeishInterface|Commit|string $arg
+     * @param array $options
+     *
+     * @throws \RuntimeException
+     * @return string
+     */
+    public function reset($arg = null, Array $options = array())
```
Comment (matteosister) [inline src/GitElephant/Command/ResetCommand.php]: could you please lowercase Array?

**Comment by matteosister** [comment]
awesome, thanks, jsut fix the Array case and it's perfect :+1:

**Comment by programaths** [comment]
Hi,

it has been changed to comply :-)

**Comment by matteosister** [comment]
thanks! :heart:

**Comment by programaths** [comment]
You are welcome ! And the library is very useful !

**Pull request merged**